server/src/controllers/google_home.rs

```diff
@@ -0,0 +1,325 @@
+                    ),
+                    (
+                        ServiceName::HumiditySensor,
+                        CharacteristicName::CurrentHumidity,
+                    ),
+                ],
+                accessory::Type::Houseflow(_) => todo!(),
+                _ => todo!(),
+            };
+            let futures = values
+                .into_iter()
+                .map(|(service_name, characteristic_name)| {
+                    master_provider.read_characteristic(
+                        accessory_id,
+                        service_name.to_owned(),
+                        characteristic_name.to_owned(),
+                    )
+                });
+            let characteristics = join_all(futures).await;
+            let mut state = query::response::State {
+                online: true,
+                ..Default::default()
+            };
+
+            for characteristic in characteristics {
+                match characteristic {
+                    Ok(Characteristic::CurrentTemperature(
+                        characteristics::CurrentTemperature { temperature },
+                    )) => {
+                        state.thermostat_temperature_ambient = Some(temperature as f64);
+                    }
+                    Ok(Characteristic::CurrentHumidity(characteristics::CurrentHumidity {
+                        humidity,
+                    })) => {
+                        // state.thermostat_humidity_ambient = Some(humidity as f64);
+                    }
+                    Err(err) => {
+                        return Ok((
+                            device.id.clone(),
+                            query::response::PayloadDevice {
+                                status: query::response::PayloadDeviceStatus::Error,
+                                error_code: Some(err.to_string()),
+                                state,
+                            },
+                        ))
+                    }
+                    _ => todo!(),
+                };
+            }
+
+            Ok((
+                device.id.clone(),
+                query::response::PayloadDevice {
+                    status: query::response::PayloadDeviceStatus::Success,
+                    error_code: None,
+                    state,
+                },
+            ))
+        }
+    });
+
+    Ok(query::response::Payload {
+        error_code: None,
+        debug_string: None,
+        devices: futures::future::try_join_all(responses)
+            .await?
+            .into_iter()
+            .collect(),
+    })
+}
+
+pub async fn execute(
+    master_provider: providers::MasterHandle,
+    user_id: user::ID,
+    payload: &execute::request::Payload,
+) -> Result<execute::response::Payload, ServerError> {
+    todo!()
+    // let requests = payload
+    //     .commands
+    //     .iter()
+    //     .flat_map(|cmd| cmd.execution.iter().zip(cmd.devices.iter()));
```

Comment:
zip is not what you want here, it will match up corresponding pairs (so the first execution will match with the first device, the second execution with the second device and so on), but you need to take the cartesian product instead (so that every execution gets runs for every device).